modules/fedimint-mint-client/src/lib.rs

```diff
@@ -1526,7 +1600,6 @@ pub enum MintClientStateMachines {
     Output(MintOutputStateMachine),
     Input(MintInputStateMachine),
     OOB(MintOOBStateMachine),
-    Restore(MintRestoreStateMachine),
```

Comment:
I should have caught this, running recoveries or logs of past recoveries could crash the client. We can:
1. Revert this change and put a dummy struct in there (since enums are length-encoded we don't have to preserver `MintRestoreStateMachine` but can just ignore whatever bytes are found)
2. Write a client DB migration that removes this particular SM. This would be super hacky since it would have to be done on the client and not module level.
3. Wait for SM migrations to land. What's the progress looking there @m1sterc001guy?